pkg/graveler/ref/resolve_ref.go

```diff
@@ -126,8 +136,11 @@ func revResolveAHash(ctx context.Context, store Store, addressProvider ident.Add
 	}
 	commitID := graveler.CommitID(addressProvider.ContentAddress(commit))
 	return &graveler.ResolvedRef{
-		Type:     graveler.ReferenceTypeCommit,
-		CommitID: commitID,
+		Type: graveler.ReferenceTypeCommit,
+		BranchRecord: graveler.BranchRecord{
+			Branch: &graveler.Branch{
+				CommitID: commitID,
+			}},
```

Comment:
I know this might be a little be confusing but I think that the alternative of adding all the fields explicitly is worse.
See comment below - It is confusing only in the initialization of the commit ref and we can add a documentation here, WDYT?